src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -2108,7 +2360,9 @@
    77 add-catalog-inputs-pkey
    78 add-catalog-inputs-hash
    79 add-report-partition-indexes-on-certname-end-time
-   80 add-workspaces-tables})
+   80 add-workspaces-tables
+   81 migrate-resource-events-to-declarative-partitioning
```

Comment:
Is anyone who would be using these migrations guaranteed to be on at least Postgresql 11?

Reply:
Separate question; are the migrations wrapped in transactions? I feel like these two should be.

Reply:
I'm pretty sure these migration would fail on anything less than 11 due to lack of declarative partitioning, so, yeah, the question is how do we want to handle that for FOSS upgrades?

Reply:
After chatting with Austin, will update oldest-allowed-db to 11.0 to block upgrades from 9.6 which doesn't support declarative partitions (and from 10 which is no longer supported by postgres).